keras/layers/core.py

```diff
@@ -566,6 +566,8 @@ def get_output_shape_for(self, input_shape):
                 else:
                     return K.int_shape(x)
             # otherwise, we default to the input shape
+            warnings.warn("Output shape not specified for layer {} and inferring doesn't work on Theano. "
+                          "Defaulting to {}".format(self.name, input_shape))
```

Comment:
That's a good warning to have. Reformat: `"` should be `'`.
"`output_shape`  argument not specified for layer {} and cannot be automatically inferred with the Theano backend. Defaulting to output shape {} (same as input shape). If the expected output shape is different, specify it via the `output_shape` argument. "